### PR TITLE
qemu: Add --qemu-firmware bios|uefi|uefi-secure

### PR DESCRIPTION
**Pull request by cgwalters**
More moving of code into mantle that was in coreos-assembler.
We want to test UEFI and Secure Boot for obvious reasons.

### CODE REVIEW

**Comment by cgwalters** [comment]
See https://github.com/coreos/coreos-assembler/pull/570

File: platform/qemu.go [hunk 1]
```diff
@@ -403,6 +406,35 @@ func baseQemuArgs(board string) []string {
 	}
 }
 
+func (builder *QemuBuilder) setupUefi(secureBoot bool) error {
+	varsVariant := ""
+	if secureBoot {
+		varsVariant = ".secboot"
+	}
+	varsSrc, err := os.Open(fmt.Sprintf("/usr/share/edk2/ovmf/OVMF_VARS%s.fd", varsVariant))
```
Comment (arithx) [inline platform/qemu.go]: I'm not too familiar with this, is there a potential issue with having multiple spawned machines all touching this file?

Comment (cgwalters) [reply platform/qemu.go]: We're making a private writable copy; this line is just opening it read-onyl.

**Comment by cgwalters** [comment]
My test case for this was:

```
$ cosa kola run -- --qemu-firmware uefi-secure
$ grep lockdown tmp/kola/basic/8130194a-e7d8-4cde-a8a6-3c96153b4b66/console.txt
[    0.000000] Kernel is locked down from EFI Secure Boot mode; see man kernel_lockdown.7
...
```

File: platform/machine/unprivqemu/cluster.go [hunk 2]
```diff
@@ -94,6 +94,7 @@ func (qc *Cluster) NewMachineWithOptions(userdata *conf.UserData, options platfo
 	builder := platform.NewBuilder(board, confPath)
 	defer builder.Close()
 	builder.Uuid = qm.id
+	builder.Firmware = qc.flight.opts.Firmware
```
Comment (arithx) [inline platform/machine/unprivqemu/cluster.go]: Unrelated to this PR but are we missing the setting of `builder.Swtpm`? It's being defaulted to `true` in `platform.NewBuilder` and a cursory scan didn't see it being set to the value of `qc.flight.opts.Swtpm` anywhere.

Comment (cgwalters) [reply platform/machine/unprivqemu/cluster.go]: I think we should nuke that option and just set it off if !x86_64 - will do it in a followup PR.

**Pull request merged**